plugin-dev/Source/Sentry/Sentry.Build.cs

```diff
@@ -1,16 +1,422 @@
+		                " -S \""+m_targetLocation+"\""+
+		                " -B \""+buildDirectory+"\""+
+		                buildToolchain+
+		                " -DCMAKE_BUILD_TYPE="+GetBuildType(target)+
+		                " -DCMAKE_INSTALL_PREFIX=\""+installPath+"\""+
+		                options+
+		                " "+m_cmakeArgs;
+
+		Console.WriteLine("CMakeTarget calling cmake with: "+arguments);
+
+		return program+arguments;
+	}
+
+    private string CreateCMakeBuildCommand(string buildDirectory, string buildType)
+    {
+        return GetCMakeExe()+" --build \""+buildDirectory+"\" --config "+buildType;
+    }
+
+    private string CreateCMakeInstallCommand(string buildDirectory, string buildType)
+    {
+        return GetCMakeExe()+" --build \""+buildDirectory+"\" --target install --config "+buildType;
+    }
+
+    private Tuple<string, string> GetExecuteCommandSync()
+    {
+        string cmd = "";
+        string options = "";
+
+        if((BuildHostPlatform.Current.Platform == UnrealTargetPlatform.Win64) 
+#if !UE_5_0_OR_LATER
+            || (BuildHostPlatform.Current.Platform == UnrealTargetPlatform.Win32)
+#endif//!UE_5_0_OR_LATER
+            )
+        {
+            cmd="cmd.exe";
+            options="/c ";
+        }
+        else if(IsUnixPlatform(BuildHostPlatform.Current.Platform)) 
+        {
+            cmd="bash";
+            options="-c ";
+        }
+        return Tuple.Create(cmd, options);
+    }
+
+    private int ExecuteCommandSync(string command)
+    {
+        var cmdInfo=GetExecuteCommandSync();
+
+        if(IsUnixPlatform(BuildHostPlatform.Current.Platform)) 
+        {
+            command=" \""+command.Replace("\"", "\\\"")+" \"";
+        }
+
+        Console.WriteLine("Calling: "+cmdInfo.Item1+" "+cmdInfo.Item2+command);
+
+        var processInfo = new ProcessStartInfo(cmdInfo.Item1, cmdInfo.Item2+command)
+        {
+            CreateNoWindow=true,
+            UseShellExecute=false,
+            RedirectStandardError=true,
+            RedirectStandardOutput=true,
+            WorkingDirectory=m_modulePath
+        };
+
+        StringBuilder outputString = new StringBuilder();
+        Process p = Process.Start(processInfo);
+
+        p.OutputDataReceived+=(sender, args) => {outputString.Append(args.Data); Console.WriteLine(args.Data);};
+        p.ErrorDataReceived+=(sender, args) => {outputString.Append(args.Data); Console.WriteLine(args.Data);};
+        p.BeginOutputReadLine();
+        p.BeginErrorReadLine();
+        p.WaitForExit();
+
+        if(p.ExitCode != 0)
+        {
+             Console.WriteLine(outputString);
+        }
+        return p.ExitCode;
+    }
+
+    private bool IsUnixPlatform(UnrealTargetPlatform platform) {
+        return platform == UnrealTargetPlatform.Linux || platform == UnrealTargetPlatform.Mac;
+    }
+}
+
 public class Sentry : ModuleRules
 {
+	[CommandLine("-usebreakpad")]
+	public bool bUseBreakpad = false;
```

Comment:
If I say `false`, what am I using then?
And would that be true regardless of platform? (On Xbox today we only support breakpad so that's odd)

Seems like config will be more complicated than a `bool` as it'll depend on what platform you're targeting, and what's supported there.

Reply:
I should probably rename this. Its an override more then anything. Native already select the backend needed for selected platforms but this allows the build system to override to build to use breakpad on windows. its needed for shipping releases as we only ship breakpad for the marketplace version

Reply:
Yeah then it needs a better name, and docs on it (XML docs over C# members)